### DISCUSSION TITLE
Questions on toys and fit biases

### DISCUSSION THREAD

**Discussion by MarcelHoh** [scikit-hep/pyhf#2251] [Q&A]
Dear experts,

I'm using pyhf for my B ->  Xu l nu via weak annihilation study at Belle II. I'm running some toy studies and observe some biases in the fits. I've uploaded a small spec [here](https://gist.github.com/MarcelHoh/73e2b8eb149bf050acc591927f989b7c). 

[Edit: my question on toys was answered here: https://hep-physics.rockefeller.edu/~luc/technical_reports/cdf5776_pulls.pdf] 

Running my fit on the toys, I observe biases which seem to be related to certain nuisance parameters. In the spec attached in the gist I have included just one nuisance parameter, `ff_Omega[3]` which is one such problematic parameter. Generating 10000 toys and plotting the pulls I find an odd structure:
    
![image](https://github.com/scikit-hep/pyhf/assets/58096701/f585a3a6-2d3a-42ae-bcf6-72e8f8e82958)

Which seems to have caused biases in my two parameters of interest, `mu_wa_charged`, and `mu_wa_mixed`. 

![image](https://github.com/scikit-hep/pyhf/assets/58096701/fbb868c2-0074-461d-9fb1-7c0d6d12b201)
![image](https://github.com/scikit-hep/pyhf/assets/58096701/e9d06c9c-76c3-4416-b85a-e1cac62c15b0)

I'm a little lost as to what causes this bias. The change in shape to the `Xulnu` template caused by the nuisance parameter is similar to the shape of the `WA_Charged/Mixed` samples but I would expect this to just be associated with a large uncertainty as the fit cannot distinguish between the weak annihilation events and changes to this form factor.
I would also naively expect the structure in the `ff_Omega[3]` pulls to be somewhat mirrored in the weak annihilation yields rather than a biased but still normal distribution.

I'm unsure where to go from here, any tips or tricks to fix this bias would be appreciated!
Thanks in advance!

Cheers,
Marcel

Edit:
To add some more information, the fitted central values look correct, like a correlated multi-normal distribution as expected. The errors however seem to group into two clusters, depending on the fitted value of the `ff_Omega[3]` nuisance parameter.

![image](https://github.com/scikit-hep/pyhf/assets/58096701/a56fa871-abe8-4977-9550-563dbbe44431)
![image](https://github.com/scikit-hep/pyhf/assets/58096701/b27c7923-0e20-467d-9ed8-163ac1394927)
![image](https://github.com/scikit-hep/pyhf/assets/58096701/1c60e28f-f4d4-409a-be18-5e7925842c54)
![image](https://github.com/scikit-hep/pyhf/assets/58096701/f00bd84f-09cd-4cae-9d28-98df56f217dd)

**Comment by kratsg** [comment]
Did this get answered by https://hep-physics.rockefeller.edu/~luc/technical_reports/cdf5776_pulls.pdf? Can you elaborate more - so we can mark as answered?
